### DISCUSSION TITLE
config warning: Unknown CompileFlags key 'Compiler'

### DISCUSSION THREAD

**Discussion by asceking** [clangd/clangd#1223] [Q&A] [answered]
When I try to change the user configuration file, a warning is reported: Unknown CompileFlags key 'Compiler'
[{
	"resource": "/home/asceking/.config/clangd/config.yaml",
	"owner": "_generated_diagnostic_collection_name_#8",
	"severity": 4,
	"message": "Unknown CompileFlags key 'Compiler'",
	"source": "clangd-config",
	"startLineNumber": 2,
	"startColumn": 3,
	"endLineNumber": 2,
	"endColumn": 11
}]
I just write these things to the configuration file:
`CompileFlags:
  Compiler: g++
  Add: [-Wall, -Wextra]
`
so what's going wrong?

![clangd-warning](https://user-images.githubusercontent.com/71561494/180466160-3b3d96e8-92f5-4e8b-9544-6c7d9d2bf969.png)

`clangd-12 --version `:
Ubuntu clangd version 12.0.1-19ubuntu3

**Comment by HighCommander4** [comment, marked as answer]
`Compiler` was added in clangd 14. (If you look at the [documentation page](https://clangd.llvm.org/config.html#compiler) for config options, the version in which they were added is shown on the right-hand side).

Additionally, in my experience, `Compiler` [is buggy](https://github.com/clangd/clangd/issues/1089#issuecomment-1087226248) even in clangd 14.

If you provide some more details about what you're hoping to accomplish by using `Compiler` I may be able to suggest alternatives.

**Reply by asceking** [reply]
Many thanks. And I'm sorry that I was so careless that I ignored the version marked in the documentation.
